src/borg/testsuite/archiver/create_cmd.py

```diff
@@ -500,6 +501,7 @@ def test_repeated_files(self):
         self.cmd(f"--repo={self.repository_location}", "create", "test", "input", "input")
 
     @pytest.mark.skipif("BORG_TESTS_IGNORE_MODES" in os.environ, reason="modes unreliable")
+    @pytest.mark.skipif(is_win32, reason="modes unavailable on Windows")
```

Comment:
guess on win32, we'ld have to look at `st.st_file_attributes` instead of `.st_mode`.

is `.st_mode` completely unused on win32?

Reply:
i checked: st_mode somehow "works" on win32, too:

usually, files are -rw-rw-rw and dirs are drwxrwxrwx.

so guess the **reason** should say "umask unavailable on Windows".

Reply:
st_mode is not real on NTFS (or the windows kernel in general), but the urct (C runtime library) does kind of emulate it based on FILE_ATTRIBUTE_DIRECTORY, FILE_ATTRIBUTE_READONLY, and checking the filename for ".exe", ".cmd", ".bat", and ".com".

Which produces vaguely sensible results